### PR TITLE
Commit init #1 task 10

### PR DESCRIPTION
**Pull request by AndrewsSkBaron**

### CODE REVIEW

File: src/main/resources/andrew-cart.json [hunk 1]
```diff
@@ -1 +1 @@
-{"cartName":"andrew-cart","realItems":[{"weight":1560.0,"name":"Audi","price":32026.9}],"virtualItems":[{"sizeOnDisk":20000.0,"name":"Windows","price":11.0}],"total":38445.479999999996}
```
Comment (KirKarpov) [inline src/main/resources/andrew-cart.json:1]: Initial cart object shouldn't be changed. Why do need to remove data from it?

File: src/test/java/MainTest.java [hunk 2]
```diff
@@ -1,24 +1,4 @@
...
-    VirtualItem disk = new VirtualItem();
-    Parser parser = new JsonParser();
-    Cart eugenCart = parser.readFromFile(new File("src/main/resources/eugen-cart.json"));
-
-    private Main main;
-
-    @Before
-    public void start(){
 
-    }
```
Comment (KirKarpov) [inline src/test/java/MainTest.java:23]: If I am not mistaken it is also initial file. It shouldn't be changed

File: src/test/java/parser/JsonParserTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import java.io.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class JsonParserTest {
+    private Cart cart = new Cart("test-andrew-cart");
+    private Gson gson = new Gson();
+
+    @Test
+    void readFromFile() throws IOException {
```
Comment (KirKarpov) [inline src/test/java/parser/JsonParserTest.java:40]: Could you please explain what do you test? As of now I see that you are create file with no data in it. Also why you are using the same name of the file that already created ?

File: src/test/java/parser/JsonParserTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        writer.close();
+
+        JsonParser jsonParser = new JsonParser();
+        Cart cartObj = jsonParser.readFromFile(new File("src/main/resources/" + cart.getCartName() + ".json"));
+        assertEquals(cartObj.getCartName(), cart.getCartName());
+        assertEquals(cartObj.getTotalPrice(), cart.getTotalPrice());
+    }
+
+    @Test
+    void writeToFile() throws IOException {
```
Comment (KirKarpov) [inline src/test/java/parser/JsonParserTest.java:51]: Your test doesn't test read from file method

File: src/test/java/parser/JsonParserTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        gson.fromJson(reader.readLine(), Cart.class);
+        reader.close();
+        JsonParser jsonParser = new JsonParser();
+        Cart cartObj = jsonParser.readFromFile(new File("src/main/resources/" + cart.getCartName() + ".json"));
+        assertEquals(cartObj.getCartName(), cart.getCartName());
+        assertEquals(cartObj.getTotalPrice(), cart.getTotalPrice());
+    }
+
+    @Test
+    public void testThrowsException() throws NoSuchFileException {
```
Comment (KirKarpov) [inline src/test/java/parser/JsonParserTest.java]: For this test please add dataset with not less than 5 parameters

File: src/test/java/shop/CartTest.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    @Test
+    void addRealItem() {
+        cart.addRealItem(car);
+        cart.addRealItem(car2);
+        assertEquals("andrew-cart", cart.getCartName());
+        assertEquals(totalCar, cart.getTotalPrice());
+    }
+
+    @Test
+    void deleteRealItem() {
```
Comment (KirKarpov) [inline src/test/java/shop/CartTest.java:48]: If you are removing element why are you checking price with removed element?

File: src/test/java/shop/CartTest.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    @Test
+    void addVirtualItem() {
+        cart.addVirtualItem(disk);
+        cart.addVirtualItem(disk2);
+        assertEquals("andrew-cart", cart.getCartName());
+        assertEquals(totalDisk, cart.getTotalPrice());
+    }
+
+    @Test
+    void deleteVirtualItem() {
```
Comment (KirKarpov) [inline src/test/java/shop/CartTest.java:65]: same there

**Review by KirKarpov — CHANGES_REQUESTED** [review]

**Comment by AndrewsSkBaron** [comment]
кукку

**Comment by AndrewsSkBaron** [comment]
Duplicate of #2

**Pull request closed without merging**